crates/hir_def/src/type_ref.rs

```diff
@@ -89,7 +89,7 @@ pub enum TypeRef {
     Array(Box<TypeRef>, ConstScalar),
     Slice(Box<TypeRef>),
     /// A fn pointer. Last element of the vector is the return type.
```

Comment:
Using the last element for the return type makes for a slightly awkward API. I wouldn't mind changing this to give the return type its specific field, if you feel that's valuable.

Reply:
sounds resoanable to change this and see if it simplifies stuff, though looking at things this only seems to be a bit bothersome in hir rendering where I added a comment about how to possibly simplify that already, so sticking with what we have here is probably good